core/src/main/java/io/grpc/ClientCall.java

```diff
@@ -203,4 +203,16 @@ public boolean isReady() {
   public void setMessageCompression(boolean enabled) {
     // noop
   }
+
+  /**
+   * Sets the compression algorithm for this call.  If the server does not support the compression
```

Comment:
I don't think we can fail the call when the compression algorithm is unsupported within `DelayedStream`.